src/lib/Model/Suggestion/SuggestionCollection.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Search\Model\Suggestion;
+
+use Ibexa\Contracts\Core\Collection\MutableArrayList;
+
+/**
+ * @template-extends \Ibexa\Contracts\Core\Collection\MutableArrayList<\Ibexa\Search\Model\Suggestion\Suggestion>
+ */
+final class SuggestionCollection extends MutableArrayList
+{
+    /**
+     * @param mixed $item
+     */
+    public function append($item): void
+    {
+        if (!$item instanceof Suggestion) {
+            throw new \TypeError(
```

Comment:
I think you should not use built-in PHP type throwables. Use `LogicException` instead, if it's a logical issue to pass an incorrect item type. Or `InvalidArgumentException`, maybe even the one from Ibexa Contracts namespace.